eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Mempool.hs

```diff
@@ -100,24 +102,42 @@ mempoolTransition ::
 mempoolTransition = do
   TRC trc@(_ledgerEnv, ledgerState, tx) <-
     judgmentContext
+
   -- This rule only gets invoked on transactions within the mempool.
   -- Add checks here that sanitize undesired transactions.
+
+  let whenFailureFree' action = ifFailureFree action (pure ledgerState)
```

Comment:
I think it would be nice to add such function to the `small-steps` instead of having just one here locally (also I hate ticks :smile:):
```haskell
whenFailureFreeDefault :: a -> Rule sts rtype a -> Rule sts rtype a
whenFailureFreeDefault defValOnFailure actionOnNoFailure =
  ifFailureFree actionOnNoFailure (pure defValOnFailure)
```

Reply:
Great idea!